testutil-server/src/main/java/io/spine/testing/server/blackbox/BlackBoxInput.java

```diff
@@ -0,0 +1,174 @@
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.testing.server.blackbox;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.Any;
+import com.google.protobuf.Message;
+import io.spine.base.CommandMessage;
+import io.spine.base.EventMessage;
+import io.spine.base.Identifier;
+import io.spine.core.Ack;
+import io.spine.core.Command;
+import io.spine.core.Event;
+import io.spine.grpc.MemoizingObserver;
+import io.spine.server.BoundedContext;
+import io.spine.server.aggregate.ImportBus;
+import io.spine.server.commandbus.CommandBus;
+import io.spine.server.event.EventBus;
+import io.spine.testing.client.TestActorRequestFactory;
+import io.spine.testing.server.TestEventFactory;
+
+import java.util.Collection;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Lists.asList;
+import static com.google.common.collect.Lists.newArrayListWithCapacity;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * An input of a {@link BlackBoxBoundedContext}, which receives domain messages.
+ */
+@VisibleForTesting
+final class BlackBoxInput {
+
+    private final CommandBus commandBus;
+    private final EventBus eventBus;
+    private final ImportBus importBus;
+    private final TestActorRequestFactory requestFactory;
+    private final TestEventFactory eventFactory;
+    private final MemoizingObserver<Ack> observer;
+
+    BlackBoxInput(BoundedContext boundedContext,
+                  TestActorRequestFactory requestFactory,
+                  MemoizingObserver<Ack> observer) {
+        this.commandBus = boundedContext.getCommandBus();
+        this.eventBus = boundedContext.getEventBus();
+        this.importBus = boundedContext.getImportBus();
+        this.requestFactory = checkNotNull(requestFactory);
+        this.eventFactory = eventFactory(requestFactory);
+        this.observer = checkNotNull(observer);
+    }
+
+    /**
+     * Sends off a provided command to the Bounded Context.
```

Comment:
Well, it is weird. The method is `receivesCommands`, but you say `sends`.

Also, a method is about multiple commands.